File: Kth Largest Element in a Stream/Kth Largest Element in a Stream.py
# link :https://leetcode.com/problems/kth-largest-element-in-a-stream/description/

class KthLargest:
    def __init__(self, k: int, nums: List[int]):
        self.pq = []
        self.K = k
        for val in nums:
            self.add(val)

    def add(self, val: int) -> int:
        heapq.heappush(self.pq, val)
        if len(self.pq) > self.K:
            heapq.heappop(self.pq)
        return self.pq[0]

File: Kth Largest Element in a Stream/Kth Largest Element in a Stream.cpp
// link : https://leetcode.com/problems/kth-largest-element-in-a-stream/description/
class KthLargest
{
    priority_queue<int, vector<int>, greater<int>> pq;
    int K;

public:
    KthLargest(int k, vector<int> &nums)
    {
        K = k;
        for (auto it : nums)
        {
            pq.push(it);
            if (pq.size() > K)
            {
                pq.pop();
            }
        }
    }

    int add(int val)
    {
        pq.push(val);
        if (pq.size() > K)
        {
            pq.pop();
        }
        return pq.top();
    }
};
